rust_src/src/frames.rs

```diff
@@ -195,6 +208,115 @@ impl From<LispFrameLiveOrSelected> for LispFrameRef {
     }
 }
 
+// Perform the switch to frame FRAME.
+//
+// If FRAME is a switch-frame event `(switch-frame FRAME1)', use
+// FRAME1 as frame.
+//
+// If TRACK is non-zero and the frame that currently has the focus
+// redirects its focus to the selected frame, redirect that focused
+// frame's focus to FRAME instead.
+//
+// FOR_DELETION non-zero means that the selected frame is being
+// deleted, which includes the possibility that the frame's terminal
+// is dead.
+//
+// The value of NORECORD is passed as argument to Fselect_window.
+#[no_mangle]
+pub extern "C" fn do_switch_frame(
+    mut frame: LispObject,
+    track: i32,
+    for_deletion: i32,
+    norecord: LispObject,
+) -> LispObject {
+    let sf: LispFrameRef = selected_frame();
+
+    // If FRAME is a switch-frame event, extract the frame we should
+    // switch to.
+    // TODO: this could be probably better
+    if let Some(event) = frame.as_cons() {
+        let (head, tail) = event.into();
+        if head.eq(Qswitch_frame) {
+            if let Some(f) = tail.as_cons() {
+                frame = f.car().into();
+            }
+        }
+    }
+
+    // This used to say CHECK_LIVE_FRAME, but apparently it's possible for
```

Comment:
We should update comments as needed to keep them relevant to the Rust port. Since `CHECK_LIVE_FRAME` is no longer used this could say "This used to check for a live frame,....".